tests/test_json.py

```diff
@@ -170,6 +170,29 @@ class Config:
     assert m.json() == '{"x": "P0DT0H2M3.000000S"}'
 
 
+def test_con_decimal_encode() -> None:
+    """
+    Makes sure a decimal with decimal_places = 0, as well as one with places
+    can handle a encode/decode roundtrip.
+    """
+
+    class Id(ConstrainedDecimal):
+        max_digits = 22
+        decimal_places = 0
+        ge = 0
+
+    ObjId = Id
+
+    class Obj(BaseModel):
+        id: ObjId
+        name: str
+        price: Decimal = Decimal('0.01')
+
+    test_obj = Obj(id=1, name='Test Obj')
+    cycled_obj = Obj.parse_raw(test_obj.json())
+    assert test_obj == cycled_obj
```

Comment:
This is smart, but not the cleanest way of writing tests. Tests should demonstrate what they're checking explicitly.

Better to do something like

```py
assert Obj(id=1).json() == '{"id": 1, "price": 0.01}'
```

Reply:
I wanted to include this since the problematic behaviour were strictly related to encoding and then parsing an object. 
But I'll change it if you prefer the other way. However it doesn't capture that parsing the object back also works so maybe add a second parse as well?

Using your method that could look like this.
```python
assert Obj(id=1) == Obj.parse_raw('{"id": 1, "price": " 0.01}')
```